packages/core/admin/server/services/__tests__/api-token.test.js

```diff
@@ -717,6 +781,45 @@ describe('API Token', () => {
     });
   });
 
+  test('Updates permissions field of a custom token with unknown permissions', async () => {
```

Comment:
This test fails. I think it's mixing two ideas:
1) starting with invalid permissions and updating to valid permissions should work
2) updating with invalid permissions should throw

Reply:
Ah, ok. That's something we should test though: if a token has invalid permissions attached (like a content type that was removed), and an update is performed on non-permissions fields, it should still work without error (and clean the invalid permissions?)

Reply:
https://github.com/strapi/strapi/pull/14248 would handle that scenario

But we can still add a test just for the sake of it :slightly_smiling_face:

Reply:
That one is good enough for me, considering content types can't be updated without a restart there shouldn't be any way it could happen.